### PR TITLE
New features added to Projview

### PR DESCRIPTION
**Pull request by trygvels**
Hi!
I have added some missing features to the projview function. 

The new additions are
- subplots
- remove mono-dipole
- top left and right labels
- preserving tickmarkers when specifying ticklabels
- fontstyle
- tickdirection and vertical label rotation
- format argument now used in labeling

I have also made som minor changes to existing functionality, but its up to you if you want to keep it
- Adjusted rotated graticule to match standard graticule
- Added a width argument, because overriding_plot_properties was a bit of a hassle

An example of the new features is:
```Python
hp.projview(dust, rlabel=r"A$_{\mathsf{dust}}$", llabel=r"$Q$", unit=r"$\mu$K", fontname="serif", min=-100, max=100, width=10, show_tickmarkers=True, cbar_ticks=[-100,-50, 0,50, 100], cb_orientation="vertical", sub=221, override_plot_properties={"cbar_tick_direction":"in"})
hp.projview(dust, norm="symlog", rlabel=r"A$_{\mathsf{dust}}$", llabel=r"$Q$", unit=r"$\mu$K", cbar_ticks=[-100,-50, 0,50, 100], remove_mono=True, show_tickmarkers=True,  cb_orientation="horizontal", sub=222, override_plot_properties={"cbar_tick_direction":"in"})
hp.projview(cmb,  rlabel=r"A$_{\mathrm{cmb}}$", llabel=r"$I$",  unit=r"$\mu$K",  min=-3e3, max=3e3,  cb_orientation="vertical", sub=223, graticule=True, )
hp.projview(cmb,  rlabel=r"A$_{\mathrm{cmb}}$", llabel=r"$I$", unit=r"$\mu$K", fontname="serif",  remove_dip=True, min=-300, max=300,  cb_orientation="horizontal", sub=224, graticule=True, rot_graticule=True, rot=(0,0,0))
```
Which produces:
example

**_Missing features:_**
- Currently, there is no way to use the same symlog functionality as mollview. I suspect this is a simple fix.
- I started implementing a coordinate rotation of the custom graticule, but ran out of time. This would be useful.

### CODE REVIEW

**Comment by zonca** [comment]
@F-Tomas would you like to review this pull request?

**Comment by F-Tomas** [comment]
I will try it out.

**Comment by trygvels** [comment]
I just added a 'symlog' normalization keyword for "norm" using matplotlib's library.
The new result for the test functions are thus:
```Python
hp.projview(dust, rlabel=r"A$_{\mathsf{dust}}$", llabel=r"$Q$", unit=r"$\mu$K", fontname="serif", min=-100, max=100, width=10, show_tickmarkers=True, cbar_ticks=[-100,-50, 0,50, 100], cb_orientation="vertical", sub=221, override_plot_properties={"cbar_tick_direction":"in"})
hp.projview(dust, norm="symlog", rlabel=r"A$_{\mathsf{dust}}$", llabel=r"$Q$", unit=r"$\mu$K", cbar_ticks=[-100,-50, 0,50, 100], remove_mono=True, show_tickmarkers=True,  cb_orientation="horizontal", sub=222, override_plot_properties={"cbar_tick_direction":"in"})
hp.projview(cmb,  rlabel=r"A$_{\mathrm{cmb}}$", llabel=r"$I$",  unit=r"$\mu$K",  min=-3e3, max=3e3,  cb_orientation="vertical", sub=223, graticule=True, )
hp.projview(cmb,  rlabel=r"A$_{\mathrm{cmb}}$", llabel=r"$I$", unit=r"$\mu$K", fontname="serif",  remove_dip=True, min=-300, max=300,  cb_orientation="horizontal", sub=224, graticule=True, rot_graticule=True, rot=(0,0,0))
```

**Comment by trygvels** [comment]
I just added another symlog normalization similar to that used by Planck, from https://github.com/zonca/paperplots i believe.
It can be used by typing norm="symlog2".
If you don't want this, just let me know and ill remove it.

**Comment by zonca** [comment]
thanks @trygvels 
does this allow to use the actual planck colormap in `projview`? that would be a nice feature!

**Comment by trygvels** [comment]
microwave_sky_reduced
Oh yes :sunglasses:

**Comment by zonca** [comment]
Nice! do you have a notebook for that I can add to the docs?

**Comment by trygvels** [comment]
Do you mean for all of the new features, or just for this one?
I can make one for the new features if you would like, if its just for this one, 
I have attached the code for this particular gif underneath, and it should work if you have the colormap stored.
```Python
    import os
    import numpy as np
    import matplotlib.pyplot as plt
    import healpy as hp
    import imageio
    import cosmoglobe as cg
    import astropy.units as u

    cmap_path = Path(data_dir.__path__[0]) / "planck_cmap_logscale.dat"
    planck_cmap = np.loadtxt(cmap_path) / 255.0
    cmap = ListedColormap(planck_cmap, "planck")

    # Get data from Cosmoglobe
    chain = cg.get_test_chain()
    sky = cg.model_from_chain(
        chain,
        nside=512,
    )
    sky.remove_dipole(30 * u.deg) # Remove monopole from CMB component

    # Plot frequencies
    filenames = []
    nus = np.logspace(np.log10(1), np.log10(800), num=50, dtype=int)
    for i, freq in enumerate(nus):
        fsky = sky(freq * u.GHz, output_unit="uK_CMB")[0].value
        img = hp.projview(
            fsky,
            cbar_ticks=[-1e3, -100, -10, 0, 10, 100, 1e3, 1e5, 1e7],
            norm="symlog2",
            cmap=cmap,
            rlabel=f"{freq} GHz",
            unit=r"$\mu K_{cmb}$",
        )

        # Save figure
        fn = f"microwavegif_{i:02d}.png"
        filenames.append(fn)
        plt.savefig(fn, bbox_inches="tight", pad_inches=0.02, dpi=70)
        plt.close()

    # build gif
    with imageio.get_writer("microwave_sky_lowres.gif", mode="I", fps=10) as writer:
        for filename in filenames:
            image = imageio.imread(filename)
            writer.append_data(image)

    # Remove files
    for filename in set(filenames):
        os.remove(filename)
```
And as you can see, i have stored the planck logscale colormap locally.
It would be _really_ nice, if this was available in healpy.
I have the .dat files with RGBs for the WMAP and standard Planck as well, if you are interested!

File: healpy/newvisufunc.py [hunk 1]
```diff
@@ -116,48 +132,62 @@ def projview(
...
-    norm : {'hist', 'log', None}
-      Color normalization, hist= histogram equalized color mapping,
-      log= logarithmic color mapping, default: None (linear color mapping)
-    kwargs : keywords
-      any additional keyword is passed to pcolormesh
+    cmap : str, optional
+        Specify the colormap. default: Viridis
+    norm : {'hist', 'log', 'symlog', 'symlog2', None}
+      Color normalization, hist= histogram equalized color mapping, log=
+      logarithmic color mapping, default: None (linear color mapping)
```
Comment (zonca) [inline healpy/newvisufunc.py:172]: what about some detail about `symlog` and `symlog2`?

File: healpy/projaxes.py [hunk 2]
```diff
@@ -886,6 +893,19 @@ def get_color_table(
...
 
+def symlog_forward(m,linthresh=1.0):
+    """
+    Alternative symmetric logarithmic function used in Planck
+    """
+    # Extra fact of 2 ln 10 makes symlog(m) = m in linear regime
+    x = m / linthresh / (2 * np.log(10))
+    return np.log10(0.5 * (x + np.sqrt(4.0 + x * x)))
+
+def symlog_backward(y,linthresh=1.0):
```
Comment (zonca) [inline healpy/projaxes.py:904]: please run the `black` formatter so we get consistent formatting

File: healpy/projaxes.py [hunk 3]
```diff
@@ -865,14 +867,19 @@ def projmap(self, map, nest=False, **kwds):
 #   http://matplotlib.org/examples/pylab_examples/custom_cmap.html
 
 
+
 def get_color_table(
-    vmin, vmax, val, cmap=None, norm=None, badcolor="gray", bgcolor="white"
+    vmin, vmax, val, cmap=None, norm=None, linthresh=1, base=10, linscale=1, badcolor="gray", bgcolor="white"
```
Comment (zonca) [inline healpy/projaxes.py:872]: I think we need a docstring here to explain `linthresh` and the other parameters. Or is there a better location?

File: healpy/projaxes.py [hunk 4]
```diff
@@ -900,7 +920,20 @@ def create_colormap(cmap, badcolor="gray", bgcolor="white"):
...
+        elif cmap=="planck_log":
+            cmap_path = os.path.join(DATAPATH, "planck_cmap_logscale.dat")
+            planck_log_cmap = np.loadtxt(cmap_path) / 255.0
+            cmap0 = matplotlib.colors.ListedColormap(planck_log_cmap, "planck_log")
+        elif cmap=="wmap":
+            cmap_path = os.path.join(DATAPATH, "wmap_cmap.dat")
+            wmap_cmap = np.loadtxt(cmap_path) / 255.0
+            cmap0 = matplotlib.colors.ListedColormap(wmap_cmap, "wmap")
+        else:
+            cmap0 = matplotlib.cm.get_cmap(cmap)
```
Comment (zonca) [inline healpy/projaxes.py:936]: nice!!

**Review by zonca — CHANGES_REQUESTED** [review]
Thanks for your contribution!

just small suggestions for improvement:

it seems there is an issue with the `newvisu` notebook, see the log on readthedocs:

https://readthedocs.org/projects/healpy/builds/16522513/

File: healpy/projaxes.py (hunk 4 above)
Comment (trygvels) [reply healpy/projaxes.py:953]: Thanks!

**Comment by trygvels** [comment]
I pushed the changes you requested now, and added a few lines to the tutorial.
I had forgotten to add the files to the manifest, but it works now! 🎉

**Review by zonca — APPROVED** [review]

**Pull request merged**